Have udp tests return idiomatic ginkgo errors, update reporting of mutating parts

jayunit100 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**


/kind cleanup

**What this PR does / why we need it**:

fixes #95216 
- by adding `ginkgo.By` directives to explain the mutations that are happening
- print the number of expected services out in the verification. 

**Which issue(s) this PR fixes**:

Fixes #95216

**Does this PR introduce a user-facing change?**:


-->
```release-note
NONE
```

jayunit100 commented on test/e2e/framework/network/utils.go:
```diff
@@ -418,8 +421,8 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
 		if eps.Equal(expectedEps) && i+1 >= minTries {
-			framework.Logf("Found all expected endpoints: %+v", eps.List())
-			return
+			framework.Logf("Found all %+v expected endpoints: %+v", len(eps.List()), eps.List())
```
added this bc in large ci runs being able to see the number makes it obvious to see that this function, for example, is being called to validate 0 endpoints (imo there's maybe a better way then calling it w/ eps=[] but we can save that debate for later date :)), for now just make the logging clear

jayunit100 commented:
@aojea 2nd part of https://github.com/kubernetes/kubernetes/pull/93837

aojea commented:
Ccan you submit a new PR for the changes to `config.DialFromNode()` , seems unrelated to this PR.
Also, related to that, there are more places using the method that need to be changed
```
[aojea@aojea-laptop kubernetes]$ grep -nr DialFromNode test/
test/e2e/network/networking.go:213:			config.DialFromNode("http", config.ClusterIP, e2enetwork.ClusterHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:215:			config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:221:			config.DialFromNode("udp", config.ClusterIP, e2enetwork.ClusterUDPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:224:			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:339:			config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:344:			config.DialFromNode("http", config.NodeIP, config.NodeHTTPPort, config.MaxTries, config.MaxTries, sets.NewString())
test/e2e/network/networking.go:351:			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, 0, config.EndpointHostnames())
test/e2e/network/networking.go:356:			config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, config.MaxTries, sets.NewString())
test/e2e/framework/network/utils.go:381:// DialFromNode executes a tcp or udp request based on protocol via kubectl exec
test/e2e/framework/network/utils.go:391:func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targetPort, maxTries, minTries int, expectedEps sets.String) {
test/e2e/common/networking.go:106:				config.DialFromNode("http", endpointPod.Status.PodIP, e2enetwork.EndpointHTTPPort, config.MaxTries, 0, sets.NewString(endpointPod.Name))
test/e2e/common/networking.go:120:				config.DialFromNode("udp", endpointPod.Status.PodIP, e2enetwork.EndpointUDPPort, config.MaxTries, 0, sets.NewString(endpointPod.Name))
```

jayunit100 commented:
> [aojea@aojea-laptop kubernetes]$ grep -nr DialFromNode test/
ha other commit got lost in translation, :) didn't know you'd review it so quickly . pushed :)

jayunit100 committed:
Update DialFromNode to return values as is done w/ other tests.  Update
comments to clarify missing probability check function (since they need
to be updated anyways b/c of the return value introduced)

jayunit100 commented:
I think they're all the same PR ? 

1) `DialFromNode` needs to return an error so the comments need updating to describe the new method signature
2) All the other tests need to now process the new DialFromNode output by failing.

aojea commented on test/e2e/framework/network/utils.go:
```diff
@@ -424,8 +427,8 @@ func (config *NetworkingTestConfig) DialFromNode(protocol, targetIP string, targ
 		if eps.Equal(expectedEps) && i+1 >= minTries {
-			framework.Logf("Found all expected endpoints: %+v", eps.List())
-			return
+			framework.Logf("Found all %+v expected endpoints: %+v", len(eps.List()), eps.List())
```
```suggestion
			framework.Logf("Found all %d expected endpoints: %+v", eps.Len(), eps.List())
```

aojea commented:
> test/e2e/framework/network/utils.go:441:20: error strings should not be capitalized or end with punctuation or a newline

jayunit100 committed:
Update test/e2e/framework/network/utils.go
add logging about min/max interval
Co-authored-by: Antonio Ojea <antonio.ojea.garcia@gmail.com>

jayunit100 commented:
updated thx @aojea

spiffxp reviewed (commented):
I'm in favor of propagating the errors but I have a nit about the comment changes

spiffxp commented on test/e2e/framework/network/utils.go:
```diff
@@ -380,17 +380,20 @@ func (config *NetworkingTestConfig) GetHTTPCodeFromTestContainer(path, targetIP
+// - if maxTries < expectedEps, this test is guaranteed to return an error, because all endpoints wont be hit.
+// - maxTries == minTries will return as soon as all endpoints succeed (or fail once maxTries is reached without
+//   success on all endpoints).
+//   In general its prudent to have a high enough level of minTries to guarantee that all pods get a fair chance at receiving traffic.
```
Given all the comment space that is spent on minTries/maxTries, would it be prudent to log those in test output?  Right now only maxTries is logged, and only on error

jayunit100 replied:
fixeddddd, added a log message below

jayunit100 commented on test/e2e/framework/network/utils.go:
```diff
@@ -380,17 +380,20 @@ func (config *NetworkingTestConfig) GetHTTPCodeFromTestContainer(path, targetIP
 }
 
-// DialFromNode executes a tcp or udp request based on protocol via kubectl exec
+// DialFromNode executes a tcp or udp curl/echo request based on protocol via kubectl exec
```
+1 will add nc

jayunit100 commented:
ok updated @spiffxp

aojea commented on test/e2e/network/networking.go:
```diff
@@ -321,39 +333,53 @@ var _ = SIGDescribe("Networking", func() {
+			ginkgo.By(fmt.Sprintf("dialing(udp) %v (node) --> %v:%v (nodeIP) and getting ZERO host endpoints", config.NodeIP, config.NodeIP, config.NodeUDPPort))
+			err = config.DialFromNode("udp", config.NodeIP, config.NodeUDPPort, config.MaxTries, config.MaxTries, sets.NewString())
+			if err != nil {
+				framework.Failf("Failure validating that node port service STOPPED removed properly: %v", err)
```
nit, it is not really stopped, the service is removed, but not a big deal, at least the message is different and will help to spot easier where the test fails

jayunit100 replied:
ok, will keep for now since its lgtm

aojea commented:
/lgtm
thanks

spiffxp reviewed (approved):
/approve
/lgtm
